tracing/src/main/java/com/palantir/tracing/TraceLocal.java

```diff
@@ -149,4 +176,30 @@ public int hashCode() {
         // identity semantics
         return super.hashCode();
     }
+
+    public static final class Builder<T> {
+        @Nullable
+        private Supplier<T> initialValue;
+
+        @Nullable
+        private TraceLocalObserver<T> observer;
+
+        private Builder() {}
+
+        @SuppressWarnings("checkstyle:HiddenField")
+        public Builder<T> initialValue(@Nonnull Supplier<T> initialValue) {
+            this.initialValue = Preconditions.checkNotNull(initialValue, "initial value supplier must not be null");
+            return this;
+        }
+
+        @SuppressWarnings("checkstyle:HiddenField")
+        public Builder<T> observer(@Nonnull TraceLocalObserver<T> observer) {
+            this.observer = Preconditions.checkNotNull(observer, "trace local observer must not be null");
```

Comment:
We generally use the following pattern to avoid hiding fields:
```suggestion
        public Builder<T> observer(@Nonnull TraceLocalObserver<T> value) {
            this.observer = Preconditions.checkNotNull(value, "trace local observer must not be null");
```

Reply:
fair enough, i just mildly preferred the argument names being non-generic